independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -69,26 +72,40 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
                         name = getDefaultName(beanClass);
                     }
                 }
-            } else if (annotation.name()
+                continue;
+            }
+            if (annotation.name()
                     .equals(DotNames.ALTERNATIVE)) {
                 isAlternative = true;
-            } else if (annotation.name()
+                continue;
+            }
+            if (annotation.name()
                     .equals(DotNames.PRIORITY)) {
                 alternativePriority = annotation.value()
                         .asInt();
-            } else {
-                if (scope == null) {
-                    scope = beanDeployment.getScope(annotation.name());
-                }
-                StereotypeInfo stereotype = beanDeployment.getStereotype(annotation.name());
-                if (stereotype != null) {
-                    stereotypes.add(stereotype);
-                }
+                continue;
+            }
+            ScopeInfo scopeAnnotation = beanDeployment.getScope(annotation.name());
+            if (scopeAnnotation != null) {
+                scopes.add(scopeAnnotation);
+                continue;
```

Comment:
Is this `continue` correct? I mean it seems like the `stereotypes` list won't be populated (which I see could be used for creating the `name`) when there is a scope.

Reply:
I think it's correct. This `continue` applies to the loop which iterates over the annotations - if an annotation is a scope than it's not a stereotype, or?

Reply:
That was going to be my next question :). I just wanted to make sure since I'm sort of fuzzy about scopes and stereotypes :)